src/rust/src/x509/verify.rs

```diff
@@ -43,6 +47,134 @@ impl CryptoOps for PyCryptoOps {
+    module = "cryptography.hazmat.bindings._rust.x509"
+)]
+struct PyServerVerifier(OwnedPolicy);
+
+impl PyServerVerifier {
+    fn as_policy(&self) -> &Policy<'_, PyCryptoOps> {
+        &self.0.borrow_dependent().0
+    }
+}
+
+#[pyo3::pymethods]
+impl PyServerVerifier {
+    #[getter]
+    fn subject<'p>(&'p self, py: pyo3::Python<'p>) -> pyo3::PyResult<Option<&'p pyo3::PyAny>> {
+        match self.0.borrow_owner() {
+            SubjectOwner::DNSName((subject, _)) => Ok(Some(subject.as_ref(py))),
+            SubjectOwner::IPAddress((subject, _)) => Ok(Some(subject.as_ref(py))),
+        }
+    }
+
+    #[getter]
+    fn validation_time<'p>(&self, py: pyo3::Python<'p>) -> pyo3::PyResult<&'p pyo3::PyAny> {
+        datetime_to_py(py, &self.as_policy().validation_time)
+    }
+}
+
+fn build_subject_owner(
+    py: pyo3::Python<'_>,
+    subject: pyo3::Py<pyo3::PyAny>,
+) -> pyo3::PyResult<SubjectOwner> {
+    let subject = subject.as_ref(py);
+
+    let x509_general_name_module =
+        py.import(pyo3::intern!(py, "cryptography.x509.general_name"))?;
+    let dns_name_class = x509_general_name_module.getattr(pyo3::intern!(py, "DNSName"))?;
+    let ip_address_class = x509_general_name_module.getattr(pyo3::intern!(py, "IPAddress"))?;
+
+    if subject.is_instance(dns_name_class)? {
+        let value = subject
+            .getattr(pyo3::intern!(py, "value"))?
+            .downcast::<pyo3::types::PyString>()?;
+
+        Ok(SubjectOwner::DNSName((
+            subject.into_py(py),
+            value.to_str()?.to_owned(),
+        )))
+    } else if subject.is_instance(ip_address_class)? {
+        let value = subject
+            .getattr(pyo3::intern!(py, "_packed"))?
+            .call0()?
+            .downcast::<pyo3::types::PyBytes>()?;
+
+        Ok(SubjectOwner::IPAddress((subject.into_py(py), value.into())))
+    } else {
+        Err(pyo3::exceptions::PyTypeError::new_err(
+            "unsupported subject type",
+        ))
+    }
+}
+
+fn build_subject<'a>(
+    py: pyo3::Python<'_>,
+    subject: &'a SubjectOwner,
+) -> pyo3::PyResult<Option<Subject<'a>>> {
+    match subject {
+        SubjectOwner::DNSName((_, dns_name)) => {
+            let dns_name = DNSName::new(dns_name)
+                .ok_or_else(|| pyo3::exceptions::PyValueError::new_err("invalid domain name"))?;
+
+            Ok(Some(Subject::DNS(dns_name)))
+        }
+        SubjectOwner::IPAddress((_, ip_addr)) => {
+            let ip_addr = IPAddress::from_bytes(ip_addr.as_bytes(py))
+                .ok_or_else(|| pyo3::exceptions::PyValueError::new_err("invalid IP address"))?;
+
+            Ok(Some(Subject::IP(ip_addr)))
+        }
+    }
+}
+
+#[pyo3::prelude::pyfunction]
+fn create_server_verifier(
+    py: pyo3::Python<'_>,
+    subject: pyo3::Py<pyo3::PyAny>,
+    time: Option<&pyo3::PyAny>,
+) -> pyo3::PyResult<PyServerVerifier> {
+    let time = match time {
+        Some(time) => py_to_datetime(py, time)?,
+        None => datetime_now(py)?,
```

Comment:
Add to docs that if not specifies, uses `now` at the time the verifier is constructed

Reply:
[de2c7b0](https://github.com/pyca/cryptography/pull/9405/commits/de2c7b0bb2933e435b0bc07e2c8cb013981e1aa0)